common/display/displayplanemanager.cpp

```diff
@@ -458,6 +460,39 @@ void DisplayPlaneManager::ValidateCachedLayers(
   *render_layers = needs_gpu_composition;
 }
 
+bool DisplayPlaneManager::ValidateSurfaceDamage(
+    const std::vector<OverlayLayer> *layers,
+    const std::vector<OverlayLayer> *previous_layers) {
+  auto layer_begin = layers->begin();
+  auto layer_end = layers->end();
+  bool validate = true;
+  size_t num_layers = layers->size();
+
+  for (size_t i = 0; i <= num_layers; i++) {
+    const HwcRegion &current_surface_damage = layers->at(i).GetSurfaceDamage();
+
+    if (current_surface_damage.kNumRects == 1) {
+      const HwcRect<int> *rect = current_surface_damage.kRects;
```

Comment:
Please find the Surfacedamage description in the https://source.android.com/devices/halref/hwcomposer2_8h_source.html

 1697  * If damage.numRects > 0, then it may be assumed that any portion of the source
 1698  * buffer not covered by one of the rects has not been modified this frame. If
 1699  * damage.numRects == 0, then the whole source buffer must be treated as if it
 1700  * has been modified.
 1701  * 
 1702  * If the layer's contents are not modified relative to the prior frame, damage
 1703  * will contain exactly one empty rect([0, 0, 0, 0]).

KnumRects = 0 then the entire frame content is changed no damage region.